be/src/runtime/collection_value.h

```diff
@@ -123,21 +123,21 @@ class CollectionValue {
     const bool* null_signs() const { return _null_signs; }
     void* mutable_data() { return _data; }
     bool* mutable_null_signs() { return _null_signs; }
-    void set_length(uint32_t length) { _length = length; }
+    void set_length(int64_t length) { _length = length; }
     void set_has_null(bool has_null) { _has_null = has_null; }
     void set_data(void* data) { _data = data; }
     void set_null_signs(bool* null_signs) { _null_signs = null_signs; }
 
 private:
     using AllocateMemFunc = std::function<uint8_t*(size_t size)>;
     static Status init_collection(CollectionValue* value, const AllocateMemFunc& allocate,
-                                  uint32_t size, PrimitiveType child_type);
+                                  int64_t size, PrimitiveType child_type);
     ArrayIterator internal_iterator(PrimitiveType child_type) const;
 
 private:
     // child column data
     void* _data;
-    uint32_t _length;
+    int64_t _length;
```

Comment:
Using `uint64_t` is better. For some places which need signed value, we use signed value there.